src/App.js

```diff
@@ -14,50 +13,90 @@ import Users from './pages/Users';
 import Header from './components/Header';
 import Dashboard from './pages/Dashboard';
 import ProjectEdit from './pages/ProjectEdit';
-import TaskDetail from './pages/TaskDetail';
-import ClientDetail from './pages/ClientDetail';
-
-function App() {
-  const LoginContainer = () => (
-    <React.Fragment>
-      <Route exact path='/' render={() => <Redirect to='/login' />} />
-      <Route exact path='/login' component={Login} />
-    </React.Fragment>
-  );
-
-  const DefaultContainer = () => (
-    <React.Fragment>
-      <Header />
-      <Switch>
-        <Route exact path='/' component={Dashboard} />
-        <Route path='/dashboard' component={Dashboard} />
-        <Route exact path='/projects' component={Project} />
-        <Route path='/projects/new' component={NewProject} />
-        <Route exact path='/projects/:id' component={ProjectDetail} />
-        <Route exact path='/projects/edit/:id' component={ProjectEdit} />
-        <Route exact path='/clients' component={Clients} />
-        <Route path='/clients/new' component={NewClient} />
-        <Route exact path='/clients/:id' component={ClientDetail} />
-        <Route path='/clients/edit' component={NewClient} />
-        <Route path='/profile' component={ProfilePage} />
-        <Route exact path='/tasks' component={Tasks} />
-        <Route exact path='/tasks/new' component={NewTask} />
-        <Route exact path='/tasks/edit/:id' component={NewTask} />
-        <Route exact path='/tasks/:id' component={TaskDetail} />
-
-        <Route exact path='/administration' component={Users} />
-      </Switch>
-    </React.Fragment>
-  );
-
-  return (
-    <BrowserRouter>
-      <Switch>
-        <Route exact path='/login' component={LoginContainer} />
-        <Route component={DefaultContainer} />
-      </Switch>
-    </BrowserRouter>
-  );
-}
+import './App.css';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isAuthenticated: this.isTokenPresent(),
+    };
+
+    this.checkAuthentication = this.checkAuthentication.bind(this);
+    this.isTokenPresent = this.isTokenPresent.bind(this);
+  }
+
+  componentDidMount() {
+    document.addEventListener('authenticated', this.checkAuthentication);
+  }
 
-export default App;
+  componentWillUnmount() {
+    document.removeEventListener('authenticated', this.checkAuthentication);
+  }
+
+  checkAuthentication() {
+    this.setState({
+      isAuthenticated: this.isTokenPresent(),
```

Comment:
can we deduct successfull authentication just from the token being present? i am not sure real companies would do it like this, because im quite sure you an easily hack that and just insert anything in this thing called token

Reply:
To hack this you need to know secret key, which is stored on backend.

Reply:
I understood what you mean, now we validate token on backend side for any request which tries to modify data (post, put, delete). Lucian said for get it's not an obligation, but we can added validation also for all get requests. In this case if you change try to access data with anything as a token - you'll get 401 error. Should I add it for GET methods too? what do you think?

Reply:
I added token validation for all requests.